examples/cpp/hello_world/HelloWorldPublisher.cpp

```diff
@@ -0,0 +1,166 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file HelloWorldPublisher.cpp
+ *
+ */
+
+#include "HelloWorldPublisher.h"
+
+#include <csignal>
+#include <stdexcept>
+#include <thread>
+
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/publisher/Publisher.hpp>
+#include <fastdds/dds/publisher/qos/DataWriterQos.hpp>
+#include <fastdds/dds/publisher/qos/PublisherQos.hpp>
+
+using namespace eprosima::fastdds::dds;
+
+std::atomic<bool> HelloWorldPublisher::stop_(false);
+
+HelloWorldPublisher::HelloWorldPublisher()
+    : participant_(nullptr)
+    , publisher_(nullptr)
+    , topic_(nullptr)
+    , writer_(nullptr)
+    , type_(new HelloWorldPubSubType())
+{
+    // Set up the data type with initial values
+    hello_.index(0);
+    hello_.message("Hello world");
+
+    // Create the participant
+    auto factory = DomainParticipantFactory::get_instance();
+    participant_ = factory->create_participant_with_default_profile();
+    if (participant_ == nullptr)
+    {
+        throw std::runtime_error("Participant initialization failed");
+    }
+
+    // Register the type
+    type_.register_type(participant_);
+
+    // Create the publisher
+    PublisherQos pub_qos = PUBLISHER_QOS_DEFAULT;
+    participant_->get_default_publisher_qos(pub_qos);
+    publisher_ = participant_->create_publisher(pub_qos, nullptr);
+    if (publisher_ == nullptr)
+    {
+        throw std::runtime_error("Publisher initialization failed");
+    }
+
+    // Create the topic
+    TopicQos topic_qos = TOPIC_QOS_DEFAULT;
+    participant_->get_default_topic_qos(topic_qos);
+    topic_ = participant_->create_topic("Hello_world_topic", "HelloWorld", topic_qos);
+    if (topic_ == nullptr)
+    {
+        throw std::runtime_error("Topic initialization failed");
+    }
+
+    // Create the data writer
+    DataWriterQos writer_qos = DATAWRITER_QOS_DEFAULT;
+    publisher_->get_default_datawriter_qos(writer_qos);
```

Comment:
Shouldn't the writer always be transient local and reliable?, probably we should fix the depth as well? Same for sub